veracruz-client/src/veracruz_client.rs

```diff
@@ -375,195 +463,57 @@ impl VeracruzClient {
-                // which is available here: https://www.itu.int/rec/T-REC-X.690-202102-I/en
-                // but it's deep inside a PDF...
-                let encoded_extension_id: [u8; 3] = [
-                    VERACRUZ_RUNTIME_HASH_EXTENSION_ID[0] * 40
-                        + VERACRUZ_RUNTIME_HASH_EXTENSION_ID[1],
-                    VERACRUZ_RUNTIME_HASH_EXTENSION_ID[2],
-                    VERACRUZ_RUNTIME_HASH_EXTENSION_ID[3],
-                ];
-                match ues.get(&encoded_extension_id[..]) {
-                    None => {
-                        error!("Our extension is not present. This should be fatal");
-
-                        Err(VeracruzClientError::RuntimeHashExtensionMissingError)
-                    }
-                    Some(data) => {
-                        info!("Certificate extension present.");
-
-                        let extension_data = data
-                            .read_all(VeracruzClientError::UnableToReadError, |input| {
-                                Ok(input.read_bytes_to_end())
-                            })?;
-
-                        info!("Certificate extension extracted correctly.");
-
-                        match self.compare_runtime_hash(extension_data.as_slice_less_safe()) {
-                            Ok(_) => {
-                                info!("Runtime hash matches.");
-
-                                Ok(())
-                            }
-                            Err(err) => {
-                                error!("Runtime hash mismatch: {}.", err);
-
-                                Err(err)
-                            }
-                        }
-                    }
-                }
-            }
-        }
-    }
-
-    /// send the data to the runtime_manager path on the Veracruz server.
-    // TODO: This function has return points scattered all over, making it very hard to follow
-    async fn send(&mut self, data: &[u8]) -> Result<Vec<u8>, VeracruzClientError> {
-        let mut enclave_session_id: u32 = 0;
-
-        if let Some(session_id) = self.remote_session_id {
-            enclave_session_id = session_id
+        let certs = self.tls_connection.peer_cert();
+        if certs.iter().count() != 1 {
+            return Err(VeracruzClientError::NoPeerCertificatesError);
         }
-
-        let mut outgoing_data_vec = Vec::new();
-        let mut start_index: usize = 0;
-        while start_index < data.len() {
-            let bytes_read = self.tls_connection.writer().write(&data[start_index..])?;
-            start_index += bytes_read;
-            let outgoing_data = Vec::new();
-            let outgoing_data_option = self.process(outgoing_data)?; // intentionally sending no data to process
-            if let Some(outgoing_data) = outgoing_data_option {
-                outgoing_data_vec.push(outgoing_data)
+        let cert = certs.iter().nth(0).unwrap().unwrap().iter().nth(0).unwrap();
```

Comment:
We should avoid `unwrap` in non-test code

Reply:
What's the best alternative?